File: src/main/java/io/github/chindeaone/collectiontracker/config/categories/overlay/MiningStatsOverlay.java
package io.github.chindeaone.collectiontracker.config.categories.overlay;

import com.google.gson.annotations.Expose;
import io.github.chindeaone.collectiontracker.config.core.Position;
import io.github.notenoughupdates.moulconfig.annotations.*;

public class MiningStatsOverlay {

    @Expose
    @ConfigOption(
            name = "Mining Stats Overlay",
            desc = "Toggles an overlay for mining stats.\n§eYou need to have mining stats (e.g. Mining Speed, Mining Fortune etc) in your stats widget for this to work.")
    @ConfigEditorBoolean
    public boolean enableMiningStatsOverlay = false;

    @Expose
    @ConfigOption(
            name = "Mining Islands Only",
            desc = "Allows the overlay to be rendered only in Mining Islands."
    )
    @ConfigEditorBoolean
    public boolean miningStatsOverlayInMiningIslandsOnly = false;

    @Expose
    @ConfigOption(
            name = "Show detailed fortune",
            desc = "Shows the fortune breakdown in the overlay."
    )
    @ConfigEditorBoolean
    public boolean showDetailedFortune = false;

    @Expose
    @ConfigLink(owner = MiningStatsOverlay.class, field = "miningStatsOverlay")
    public Position miningStatsOverlayPosition = new Position(50, 150);
}

File: src/main/java/io/github/chindeaone/collectiontracker/gui/overlays/MiningStatsOverlay.kt
package io.github.chindeaone.collectiontracker.gui.overlays

import io.github.chindeaone.collectiontracker.config.core.Position
import io.github.chindeaone.collectiontracker.config.enableMiningStatsOverlay
import io.github.chindeaone.collectiontracker.config.miningStatsPosition
import io.github.chindeaone.collectiontracker.utils.parser.MiningStatsParser

class MiningStatsOverlay : AbstractOverlay() {
    private var cachedLines: List<String> = emptyList()

    override val overlayLabel: String = "Mining Stats"

    override val position: Position get() = miningStatsPosition

    override val isEnabled: Boolean get() = enableMiningStatsOverlay

    override fun updateDimensions() {
        if (!isEnabled) return

        val lines = lines
        if (lines === cachedLines) return

        cachedLines = lines
        super.updateDimensions()
    }

    override val lines: List<String> get() = MiningStatsParser.getLines()
}
